sdcclient/_secure.py

```diff
@@ -974,3 +974,95 @@ def get_command_audit(self, id, metrics=[]):
+            List the current set of image profiles.
+
+        **Arguments**
+            - None
+
+        **Success Return Value**
+            A JSON object containing the details of each profile.
+
+        '''
+        url = "{url}/api/profiling/v1/secure/profileGroups/0/profiles".format(
+            url = self.url
+        )
+
+        res = requests.get(url, headers=self.hdrs, verify=self.ssl_verify)
+        return self._request_result(res)
+
+
+    def get_image_profile(self, profileId):
+        '''**Description**
+            Find the image profile with a (partial) profile ID <profileId> and return its json description.
+
+        **Arguments**
+            - name: the name of the image profile to fetch
+
+        **Success Return Value**
+            A JSON object containing the description of the image profile. If there is no image profile with
+            the given name, returns False. Moreover, it could happen that more than one profile IDs have a collision.
+            It is due to the fact that a partial profile ID can be passed and interpreted; in this case a set of
+            collision profiles is returned, and the full complete ID string is printed. In this case, it returns
+            false.
+
+        '''
+
+        # RETRIEVE ALL THE IMAGE PROFILES
+        ok, image_profiles = self.list_image_profiles()
+
+        if not ok:
+            return [False, self.lasterr]
+
+        
+        matched_profiles = self.__get_matched_profileIDs(profileId, image_profiles['profiles'])
+        
+        # Profile ID not found
+        if len(matched_profiles) == 0:
+            return [False, "No profile with ID {}".format(profileId)]
+        
+        # Principal workflow. Profile ID found
+        elif len(matched_profiles) == 1:
+            # Matched id. Return information
+            url = "{url}/api/profiling/v1/secure/profiles/{profileId}".format(
+                url = self.url,
+                profileId = matched_profiles[0]['profileId']
+            )
+            
+            res = requests.get(url, headers=self.hdrs, verify=self.ssl_verify)
+            return self._request_result(res)
+
+        # Collision detected. The full profile IDs are returned
+        elif len(matched_profiles) >= 2:
+            return [False, matched_profiles]
+
+
+    def __get_matched_profileIDs(self, requested_profile, profile_list):
+        '''
+        **Description**
+            Helper function for  retrieving the list of matching profile
+        
+            **Arguments**
+            - the requested profile Id (string)
+            - List of dictionary, where each discionary conatins the profile information 
+    
+        **Success Return Value**
+            List of dictionary, where each dicionary represents a profile with the ID prefix substring
+            matching the requested one
+        '''
+
+        matched_profiles = []
+
+        request_len = len(requested_profile)
+        for profile in profile_list:
```

Comment:
@flaviomutti-sysdig would you mind adding some color to this part of the code? Not knowing what the API provides, I can't easily follow the logic behind the string manipulations in the loop.

A comment and/or example of API output+function output could help future evolutions of the functionality ;-)

Thanks!

Reply:
Hi @davideschiera, thank you for your review :) I added un example for retrieving the list of image profiles and I extended the comments for describing the data structure